spark-atlas-connector/src/main/scala/com/hortonworks/spark/atlas/sql/CommandsHarvester.scala

```diff
@@ -377,4 +385,152 @@ object CommandsHarvester extends AtlasEntityUtils with Logging {
+    override def harvest(node: WriteToDataSourceV2Exec, qd: QueryDetail): Seq[AtlasEntity] = {
+      // Source table entity
+      val inputsEntities = qd.qe.sparkPlan.collectLeaves().map {
+        case h if h.getClass.getName == "org.apache.spark.sql.hive.execution.HiveTableScanExec" =>
+          Try {
+            val method = h.getClass.getMethod("relation")
+            method.setAccessible(true)
+            val relation = method.invoke(h).asInstanceOf[HiveTableRelation]
+            tableToEntities(relation.tableMeta)
+          }.getOrElse(Seq.empty)
+
+        case f: FileSourceScanExec =>
+          f.tableIdentifier.map(prepareEntities).getOrElse(
+            f.relation.location.inputFiles.map(external.pathToEntity).toSeq)
+        case HWCEntities(hwcEntities) => hwcEntities
+        case e =>
+          logWarn(s"Missing unknown leaf node: $e")
+          Seq.empty
+      }
+
+      // Supports Spark HWC (destination table entity)
+      val outputEntities = HWCEntities.getHWCEntity(node.writer)
+
+      // Creates process entity
+      val inputTablesEntities = inputsEntities.flatMap(_.headOption).toList
+      val outputTableEntities = outputEntities.toList
+      val logMap = getPlanInfo(qd)
+
+      // ML related cached object
+      if (internal.cachedObjects.contains("model_uid")) {
+        internal.updateMLProcessToEntity(inputTablesEntities, outputTableEntities, logMap)
+      } else {
+        // Creates process entity
+        val pEntity = internal.etlProcessToEntity(
+          inputTablesEntities, outputTableEntities, logMap)
+        Seq(pEntity) ++ inputsEntities.flatten ++ outputEntities
+      }
+    }
+  }
+
+  object HWCEntities extends Logging {
+    private def maybeClass(name: String): Option[Class[_]] = try {
+      Some(Class.forName(name))
+    } catch {
+      case _: ClassNotFoundException => None
+    }
+
+    private val batchReadSourceClass: Option[Class[_]] =
+      maybeClass(HWCSupport.BATCH_READ_SOURCE)
+
+    private val batchWriteClass: Option[Class[_]] = maybeClass(HWCSupport.BATCH_WRITE)
+    private val batchStreamWriteClass: Option[Class[_]] =
+      maybeClass(HWCSupport.BATCH_STREAM_WRITE)
+
+    private val streamWriteClass: Option[Class[_]] = maybeClass(HWCSupport.STREAM_WRITE)
+
+    def unapply(plan: LogicalPlan): Option[Seq[AtlasEntity]] = plan match {
+      case ds: DataSourceV2Relation
+          if ds.source.getClass.getCanonicalName.endsWith(HWCSupport.BATCH_READ_SOURCE) =>
+        Some(getHWCEntity(ds.options))
+      case _ => None
+    }
+
+    def unapply(plan: SparkPlan): Option[Seq[AtlasEntity]] = plan match {
+      case ds: DataSourceV2ScanExec
+          if ds.source.getClass.getCanonicalName.endsWith(HWCSupport.BATCH_READ_SOURCE) =>
+        Some(getHWCEntity(ds.options))
+      case _ => None
+    }
+
+    def getHWCEntity(options: Map[String, String]): Seq[AtlasEntity] = {
+      if (batchReadSourceClass.isDefined) {
+        val (db, tableName) = getDbTableNames(
```

Comment:
Currently, `options` contains the current DB, which can be set by `setDatabase`.